@xen-orchestra/web/src/components/hosts/HostsTable.vue

```diff
@@ -0,0 +1,231 @@
+import useMultiSelect from '@core/composables/table/multi-select.composable'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import { HOST_POWER_STATE } from '@vates/types'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { hosts } = defineProps<{
+  hosts: XoHost[]
+  hostReady: boolean
+  hasError: boolean
+}>()
+
+const hostsId = computed(() => hosts.map(host => host.id))
+
+const { t } = useI18n()
+const selectedHostId = useRouteQuery('id')
+const { pifsByHost, arePifsReady } = useXoPifCollection()
+const { selected, areAllSelected } = useMultiSelect(hostsId)
+
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? hostsId.value : []
+}
+
+const searchQuery = ref('')
+
+const filteredHosts = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return hosts
+  }
+
+  return hosts.filter(host => Object.values(host).some(value => String(value).toLocaleLowerCase().includes(searchTerm)))
+})
+
+const { visibleColumns, rows } = useTable('hosts', filteredHosts, {
+  rowId: record => record.id,
+  columns: define => [
+    define('checkbox', noop, { label: '', isHideable: false }),
+    define('host', record => ({ label: record.name_label, id: record.id, state: record.power_state }), {
+      label: t('name'),
+    }),
+    define('description', record => record.name_description, { label: t('description') }),
+    define(
+      'ipv4-address',
+      record => ({
+        masterIp: record.address,
+        pif:
+          pifsByHost.value
+            .get(record.id)
+            ?.map(pif => pif.ip)
+            .filter(ip => ip !== '' && ip !== record.address) ?? [],
+      }),
+      {
+        label: t('ip-addresses'),
+      }
+    ),
+    define('tags', record => record.tags, { label: t('tags') }),
+    define('more', noop, { label: '', isHideable: false }),
+  ],
+})
+
+const { pageRecords: spacesRecords, paginationBindings } = usePagination('hosts', rows)
+
+type HostHeader = 'host' | 'description' | 'ipv4-address' | 'tags'
+
+const headerIcon: Record<HostHeader, IconName> = {
+  host: 'fa:object',
+  description: 'fa:align-left',
+  'ipv4-address': 'fa:align-left',
+  tags: 'fa:square-caret-down',
+}
+
+const isHostRunning = (host: HOST_POWER_STATE) => {
+  if (host === HOST_POWER_STATE.RUNNING) {
+    return 'running'
+  } else if (host === HOST_POWER_STATE.HALTED) {
+    return 'halted'
+  } else {
+    return 'unknown'
+  }
+}
+</script>
+
+<style scoped lang="postcss">
+.host-table,
+.table-actions {
+  display: flex;
+  flex-direction: column;
+}
+
+.host-table {
+  gap: 2.4rem;
+
+  .table-actions {
+    gap: 0.8rem;
+  }
+
+  .checkbox,
+  .more {
+    width: 4.8rem;
+  }
+
+  .checkbox {
+    text-align: center;
+    line-height: 1;
+  }
```

Comment:
```suggestion
  }
  
  .ip-addresses {
    display: flex;
    justify-content: space-between;
  }
```

In Figma there is a space between addresse and +number
<img width="243" height="175" alt="image" src="https://github.com/user-attachments/assets/af467888-bbcb-4f4a-a71e-64f64bdde5a2" />

And also better for responsive

Reply:
```suggestion
.ip-addresses {
    display: flex;
    justify-content: space-between;
    align-items: center;

    .more-ips {
      color: var(--color-neutral-txt-secondary);
    }
  }
```